araig_test_runners/src/base/base_runner.py

```diff
@@ -136,13 +137,25 @@ def standardStartupSequence(self):
         # Wait until start signal received
         rospy.logwarn(rospy.get_name() + ": Waiting to start...")
         self.loopFallbackOnFlags(["start_test"])
+
         # Start received, wait for recorders to boot up. Cannot be interrupted.
-        rospy.logwarn(rospy.get_name() + ": Start received, waiting {}s for recorder init"
+        rospy.logwarn(rospy.get_name() + ": Start received, waiting max {}s for recorder init"
                         .format(self.config_param['start_logging_offset']))
-        self.sleepUninterruptedFor(self.config_param['start_logging_offset'])
+        result = self.timedLoopFallbackOnFlags(["began_recording"],self.config_param['start_logging_offset'])
```

Comment:
We could also remove this max waiting time, and keep a potentially infinite loop here that is only escapable with an interrupt.